### ISSUE TITLE
Enhancement: Add ability to autocat on developer and/or publisher

### ISSUE THREAD

**Issue by ppriest** [Theo47/depressurizer#15]
Whitelisting a selection  of publishers would be very handy. Much like the tags, filter to those where the number of games in your collection/globally is greater than 'x'.

**Comment by Theo47** [comment]
I'll see about adding something like that once i have more time.

**Comment by EnemyWithin** [comment]
I just added a basic version to my build.  It doesn't have the weighting of the Tags autocat yet.

**Comment by Theo47** [comment]
7eb6006747cb264e6b0108d6bbc680b6f64f566d

**Comment by ppriest** [comment]
Thanks for implementing this!

I can't get this to work.

The first time I tried, I think I misunderstood how it worked - I'd checked 'Owned only', regenerated the list, but then also checked 'All' to the lists, and it appears that if you do that it wasn't smart enough to realise that the 'All' lists would need to be filtered to owned games only when ran - it created >500 cats when the config indicated more like 50.

I deleted all the cats and since I've been unable to get it to function reliably - one time it created a handful of cats, but otherwise nothing.
Attached grab of config...

![image](https://cloud.githubusercontent.com/assets/2786389/13197620/29c39bd6-d7eb-11e5-9471-b89abba9bab4.png)

**Comment by EnemyWithin** [comment]
Fixed in latest commit.

**Comment by ppriest** [comment]
It works now - but I'm still a bit confused over a few points. Some of this applies to other autocats too, so they're more systemic issues:
* The filter by number of games threshold seems like it's based on all games, including hidden, but then of course when it actually comes to it, the hidden games aren't shown - so I might get a category with 1 game in it, even if the threshold is 5 because I hid 4 games from that dev/publisher.
* The list of Developers/Publishers doesn't repopulate/reload when you go back into the configuration
i.e. I select them all, then deselect one or two, close the dialog, go back in and nothing's selected again
* Selecting "All" doesn't behave as you might hope - i.e. categorising all (non-hidden) games into categories where the number of gamed per-dev/pub _that you own_ is greater than the threshold. Instead, to make use of the 'Owned only' and the threshold, you need to uncheck "All" and select all categories (which is non-ideal for two reasons - one it won't respond to new publishers/developers presumably, and two because of the above quirk around the config dialog it's a pain to maintain.

**Comment by EnemyWithin** [comment]
Good feedback, thanks!

1). Good catch.  AutoCatTags had the same "bug".  I have fixed them both to exclude hidden games.
2). This is a bug and it's fixed now.
3). You are correct.  "All" was a simple way of selecting everything (basically the same as pressing Check All, but it would include future additions).  I added the minimum game count later and didn't take "All" into account.  It's fixed now.
